### PR TITLE
Handle errant minimum value

### PR DESCRIPTION
**Pull request by imurchie**
In some cases the Safari plist gets a minimum value of -1 that gets encoded as an unsigned long, and the plist creator fails (taken from https://travis-ci.org/appium/appium-xcuitest-driver/jobs/531603055#L1394-L1413):
```
dbug iOSSim Updating Safari user settings
ERR! iOS Error setting safari preferences, prefs will not work
ERR! iOS Error: unhandled entry: 18446744073709551615
ERR! iOS     at toEntries (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:376:11)
ERR! iOS     at /Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:415:22
ERR! iOS     at Array.forEach (<anonymous>)
ERR! iOS     at toEntriesObject (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:414:21)
ERR! iOS     at toEntries (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:352:14)
ERR! iOS     at /Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:415:22
ERR! iOS     at Array.forEach (<anonymous>)
ERR! iOS     at toEntriesObject (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:414:21)
ERR! iOS     at toEntries (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:352:14)
ERR! iOS     at /Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:415:22
ERR! iOS     at Array.forEach (<anonymous>)
ERR! iOS     at toEntriesObject (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:414:21)
ERR! iOS     at toEntries (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:352:14)
ERR! iOS     at /Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:415:22
ERR! iOS     at Array.forEach (<anonymous>)
ERR! iOS     at toEntriesObject (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:414:21)
ERR! iOS     at toEntries (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:352:14)
ERR! iOS     at module.exports (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/bplist-creator/bplistCreator.js:25:17)
ERR! iOS     at Object.updatePlistFile (/Users/travis/build/appium/appium-xcuitest-driver/node_modules/appium-support/lib/plist.js:61:27)
```

### CODE REVIEW

File: lib/settings.js [hunk 1]
```diff
@@ -179,10 +179,18 @@ async function updateSafariUserSettings (sim, settingSet) {
...
     }
-    for (let [key, value] of _.toPairs(newUserSettings)) {
+    for (const [key, value] of _.toPairs(newUserSettings)) {
       userSettingSet.restrictedBool[key] = {value};
     }
 
+    // ensure that BigInt `-1` values become regular -1, so the plist creator does not fail
+    const restrictedValues = userSettingSet.restrictedValue || {};
+    for (const [key, value] of _.toPairs(restrictedValues)) {
+      if (value.rangeMinimum && value.rangeMinimum.value === 18446744073709551615n) {
```
Comment (mykola-mokhnach) [inline lib/settings.js]: should we rather fix the plist creator itself?

Comment (mykola-mokhnach) [reply lib/settings.js]: can the magic number be extracted into a constant?

**Comment by imurchie** [comment]
I can't really get a generalized solution. The problem is that, in our case, this is a signed 64 bit int that needs to be handled as such. But it could be anything.

**Comment by imurchie** [comment]
Figured out a way that seems to work in the cases I can test. We'll see.

https://github.com/joeferner/node-bplist-creator/pull/19

**Comment by imurchie** [comment]
Fixed upstream.

**Pull request closed without merging**